apps/spruce/src/pages/task/taskTabs/TaskHistory/index.tsx

```diff
@@ -89,9 +92,27 @@ const TaskHistory: React.FC<TaskHistoryProps> = ({ task }) => {
   const { pagination, tasks = [] } = taskHistory ?? {};
   const { mostRecentTaskOrder, oldestTaskOrder } = pagination ?? {};
 
-  const groupedTasks = groupTasks(tasks, shouldCollapse);
+  const [failingTest] = useQueryParam<string>(
+    TaskHistoryOptions.FailingTest,
+    "",
+  );
+  const [testFailureSearchTerm, setTestFailureSearchTerm] =
+    useState<RegExp | null>(null);
+  useEffect(() => {
+    setTestFailureSearchTerm(
+      failingTest
+        ? new RegExp(
+            validateRegexp(failingTest)
+              ? failingTest
+              : toEscapedRegex(failingTest),
+            "i",
+          )
+        : null,
+    );
+  }, [failingTest]);
```

Comment:
Does it need to be a state? Can it just be a variable (with memo if needed)? I'd like to avoid `useEffect` when possible
```suggestion
  const testFailureSearchTerm = failingTest
    ? new RegExp(
        validateRegexp(failingTest) ? failingTest : toEscapedRegex(failingTest),
        "i",
      )
    : null;
```

Reply:
Sure, that makes sense `groupedTasks` isn't memoized